socrata_util.py

```diff
@@ -0,0 +1,159 @@
+            logger: Optional parameter. Could pass in a logger object or not pass
+                in anything. If a logger object is passed in, information will be
+                logged instead of printed. If not, information will be printed.
+        """
+        self.dataset_id = dataset_id
+        self.client = socrata_client
+        if not socrata_client and socrata_params:
+            self.client = Socrata(**socrata_params)
+        self.socrata_params = socrata_params
+        self.col_dtype_dict = self.get_col_dtype_dict()
+        self.float_fields = float_fields
+        self.print_func = print
+        if logger:
+            self.print_func = logger.info
+
+    def get_col_dtype_dict(self):
+        """
+        Retrieve data dictionary of a Socrata data set in the form of a dictionary,
+        with the key being the column name and the value being the column data type
+
+    	Returns:
+    		Data dictionary of a Socrata data set in the form of a dictionary,
+            with the key being the column name and the value being the column data type.
+        """
+        dataset_col_meta = self.client.get_metadata(self.dataset_id)['columns']
+        col_dtype_dict = {col['name']: col['dataTypeName'] for col in dataset_col_meta}
+        return col_dtype_dict
+
+    def mod_dtype(self, rec, col_dtype_dict=None, float_fields=None):
+        """
+        Make sure the data type of each field in the data record matches the data type
+        of the field in the Socrata data set.
+
+    	Parameters:
+    		rec: dictionary object of the data record
+            col_dtype_dict: data dictionary of a Socrata data set in the form of a dictionary,
+            with the key being the column name and the value being the column data type
+            float_fields: list of fields that should be a float
+
+    	Returns:
+    		Dictionary object of the data record, with number, string, and boolean fields
+            modified to align with the data type of the corresponding Socrata data set.
+        """
+        col_dtype_dict = col_dtype_dict or self.col_dtype_dict
+        float_fields = float_fields or self.float_fields
+
+        identity = lambda x: x
+        dtype_func = {'number': float, 'text': str, 'checkbox': bool}
+        out = {}
+        for k,v in rec.items():
+            if k in float_fields and k in col_dtype_dict:
+                out[k] = float(v)
+            elif k in col_dtype_dict:
+                if v not in [None, '']:
+                    out[k] = dtype_func.get(col_dtype_dict.get(k, 'nonexistentKey'), identity)(v)
+        out = {k:v for k,v in out.items() if k in col_dtype_dict}
+        return out
+
+    def create_new_draft(self):
+        """
+        Create a new draft of the current dataset.
+
+        Returns:
+            Draft ID of the new draft.
+        """
+        draftDataset = requests.post('https://{}/api/views/{}/publication.json'.format(self.client.domain, self.dataset_id),
```

Comment:
Should we add error handling here or is it pretty robust? Not sure if a failed rest request will cause any cascading issues.

Reply:
Good point. This method is not used in this particular lambda. Will also work on this when we make the sandbox exporter methods importable.